### PR TITLE
Remove non-standard usage of correlation-id

### PR DESCRIPTION
**Pull request by metacubed**

### CODE REVIEW

File: src/test/java/com/ge/predix/logback/PredixEncoderTest.java [hunk 1]
```diff
@@ -226,79 +229,126 @@ public void testPredixEncoderWithNoMessageLineSeparator() throws IOException {
...
     }
 
-    private String encodeToPredixFormat(final ILoggingEvent logEvent, final String messageLineSeparator)
-            throws IOException {
+    private String encodeToPredixFormat(final ILoggingEvent logEvent,
+            final String correlationKey, final String messageLineSeparator) {
 
         PredixEncoder<ILoggingEvent> predixEncoder = new PredixEncoder<>();
         predixEncoder.setContext(loggerContext);
+        predixEncoder.setCorrelationKey(correlationKey);
```
Comment (irinaepshteyn) [inline src/test/java/com/ge/predix/logback/PredixEncoderTest.java:336]: You are testing correlation key as null (then default traceId is used) or custom.  How about empty string?

**Review by irinaepshteyn — CHANGES_REQUESTED** [review]
Done reviewing.  Please review my comments.

File: src/test/java/com/ge/predix/logback/PredixEncoderTest.java (hunk 1 above)
Comment (metacubed) [reply src/test/java/com/ge/predix/logback/PredixEncoderTest.java:360]: Added tests for empty string and blank string.

File: src/main/java/com/ge/predix/log4j2/PredixLayout.java [hunk 2]
```diff
@@ -75,7 +80,9 @@ public static PredixLayout createLayout(
                 STATUS_LOGGER.warn("Log message lines will not be separated.");
             }
         }
-        return new PredixLayout(messageLineSeparatorRegexPattern);
+        return new PredixLayout(
+                correlationKey != null ? correlationKey : DEFAULT_CORRELATION_KEY,
```
[the remark this answers is not in the document]
Comment (metacubed) [reply src/main/java/com/ge/predix/log4j2/PredixLayout.java]: done

File: src/test/java/com/ge/predix/log4j2/PredixLayoutTest.java [hunk 3]
```diff
@@ -51,14 +54,13 @@
...
     private static final String THREAD_NAME = "Thread1";
-    private static final String CORRELATION_HEADER = "X-B3-TraceId";
     private static final String ZONE_HEADER = "Zone-Id";
 
     private static final Message OBJECT_MESSAGE = SimpleMessageFactory.INSTANCE.newMessage(buildObjectMessage());
     private static final Message MULTI_LINE_TEXT_MESSAGE = SimpleMessageFactory.INSTANCE.newMessage(
             "L1\nL2" + System.lineSeparator() + "L3");
 
-    private final PredixLayout predixLayout = PredixLayout.createLayout(null);
+    private final PredixLayout predixLayout = PredixLayout.createLayout(null, null);
```
[the remark this answers is not in the document]
Comment (metacubed) [reply src/test/java/com/ge/predix/log4j2/PredixLayoutTest.java:65]: Added tests for empty string and blank string.

File: src/main/java/com/ge/predix/logback/PredixEncoder.java [hunk 4]
```diff
@@ -47,13 +47,20 @@
...
     private static final ObjectWriter JSON_WRITER = new ObjectMapper().writer();
 
     private static final int INITIAL_BUFFER_SIZE = 1024;
     private static final byte[] EMPTY_BYTES = new byte[0];
 
+    private String correlationKey = DEFAULT_CORRELATION_KEY;
     private Pattern messageLineSeparatorPattern = null;
 
+    public void setCorrelationKey(final String correlationKey) {
+        this.correlationKey = correlationKey != null ? correlationKey : DEFAULT_CORRELATION_KEY;
```
[the remark this answers is not in the document]
Comment (metacubed) [reply src/main/java/com/ge/predix/logback/PredixEncoder.java]: done

**Review by irinaepshteyn — APPROVED** [review]

**Pull request merged**